### ISSUE TITLE
Support of keyboard input for window in focus

### ISSUE THREAD

**Issue by capricorn8** [boppreh/keyboard#413]
>> Global event hook on all keyboards (captures keys regardless of focus).

While this feature is certainly of great help for certain applications, it is not what's typically required - i.e. the application in focus receives the keyboard events.

Is there any means to configure or change the event hook accordingly?

**Comment by gilbh** [comment]
I agree. I need to get the key event made when the app is in focus. Right now it triggers even when focus is on others apps, which creates new problems. Configure to work only when app in focus?

**Comment by capricorn8** [comment]
Configure to work only when app in focus?

Absolutely. I believe this is the condition for the majority of applications.

**Comment by AdnanHodzic** [comment]
Re-wrote my app with multithreading in mind so this library could catch certain keystrokes. Unfortunately, only being able to use it on "global event/focus" level makes this library unusable for me.

Truly hoping this feature can be implemented soon.

**Comment by rkcastelino** [comment]
Seconded. Would love focus-only functionality!

**Comment by capricorn8** [comment]
Me too. I have changed my existing code to using msvcrt functions, but for future and more advanced applications I would love using the sophisticated functions of this library.

**Comment by rkcastelino** [comment]
Ah! I actually just figured out a workaround based on Markus von Broady's comments [here](https://stackoverflow.com/questions/24072790/detect-key-press-in-python). It isn't optimal but it is a slight improvement.

**Comment by rkcastelino** [comment]
`def monitor():
    while True:  # making a loop
        current_window = (GetWindowText(GetForegroundWindow()))
        try:
            current_window = current_window[-10:]
        except:
            pass
        
        try:  # used try so that if user pressed other than the given key error will not be shown
            if keyboard.is_pressed('q') and current_window == 'pcPodcasts':  # if key 'q' is pressed
                quit_queue.put('quit')
                break  # finishing the loop
        except:
            break  # if user pressed a key other than the given key the loop will break`

**Comment by rkcastelino** [comment]
My keypress checking thread is as follows for reference. The script I'm currently running is always(?) going to be running in a terminal from a directory called "/.../.../pcPodcasts".

`
def monitorThread():

    while True:  # making a loop
        current_window = (GetWindowText(GetForegroundWindow()))
        try:
            current_window = current_window[-10:]
        except:
            pass

        try:  # used try so that if user pressed other than the given key error will not be shown
            if keyboard.is_pressed('q') and current_window == 'pcPodcasts':  # if key 'q' is pressed
                quit_queue.put('quit')
                break  # finishing the loop
        except:
            break  # if user pressed a key other than the given key the loop will break
`
